public/app/core/internationalization/dates.ts

```diff
@@ -20,17 +24,20 @@ export const formatDate = deepMemoize(
       return formatDate(new Date(value), format);
     }
 
-    const locale = config.locale;
-    const dateFormatter = createDateTimeFormatter(locale, format);
+    const i18n = getI18next();
+    const currentLocale = isLocaleEnabled ? config.locale : i18n.language;
```

Comment:
will this work for chinese? i think they have different language and locale codes... 🤔

Reply:
This works just fine - the browser will accept `zh-Hans` and find a suitable match.

At least, this is what we're currently doing so I think its fine to stay with it!